From e1babffe897e64cd0b210f97c5290fefe7cd90e1 Mon Sep 17 00:00:00 2001
From: Elizabeth Figura <zfigura@codeweavers.com>
Date: Mon, 7 Jul 2025 15:17:12 -0500
Subject: [PATCH 30/48] kernelbase: Open the reparse point in
 CreateDirectory().

(cherry picked from commit 3360da7066ba567924b128451d0ccaaf5a3014b4)

CW-Bug-Id: #26492
---
 dlls/kernelbase/file.c  | 2 +-
 dlls/ntdll/tests/file.c | 9 ++-------
 2 files changed, 3 insertions(+), 8 deletions(-)

diff --git a/dlls/kernelbase/file.c b/dlls/kernelbase/file.c
index b56abbac413..9ac367682ab 100644
--- a/dlls/kernelbase/file.c
+++ b/dlls/kernelbase/file.c
@@ -674,7 +674,7 @@ BOOL WINAPI DECLSPEC_HOTPATCH CreateDirectoryW( LPCWSTR path, LPSECURITY_ATTRIBU
 
     status = NtCreateFile( &handle, GENERIC_READ | SYNCHRONIZE, &attr, &io, NULL,
                            FILE_ATTRIBUTE_NORMAL, FILE_SHARE_READ, FILE_CREATE,
-                           FILE_DIRECTORY_FILE | FILE_SYNCHRONOUS_IO_NONALERT, NULL, 0 );
+                           FILE_DIRECTORY_FILE | FILE_SYNCHRONOUS_IO_NONALERT | FILE_OPEN_REPARSE_POINT, NULL, 0 );
     if (status == STATUS_SUCCESS) NtClose( handle );
 
     RtlFreeUnicodeString( &nt_name );
diff --git a/dlls/ntdll/tests/file.c b/dlls/ntdll/tests/file.c
index 6f19c9de76c..3d724859f2c 100644
--- a/dlls/ntdll/tests/file.c
+++ b/dlls/ntdll/tests/file.c
@@ -6601,13 +6601,8 @@ static void test_reparse_points(void)
      * FILE_OPEN_REPARSE_POINT under the hood. */
     swprintf( path, ARRAY_SIZE(path), L"%stestreparse_dirlink", temp_path );
     ret = CreateDirectoryW( path, NULL );
-    todo_wine ok( ret == FALSE, "got %d\n", ret );
-    todo_wine ok( GetLastError() == ERROR_ALREADY_EXISTS, "got error %lu\n", GetLastError() );
-    if (ret)
-    {
-        ret = RemoveDirectoryW( path );
-        ok( ret == TRUE, "got error %lu\n", GetLastError() );
-    }
+    ok( ret == FALSE, "got %d\n", ret );
+    ok( GetLastError() == ERROR_ALREADY_EXISTS, "got error %lu\n", GetLastError() );
 
     /* Test FindFirstFile(). */
 
-- 
2.53.0

